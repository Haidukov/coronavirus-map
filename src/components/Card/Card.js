import React from 'react';

import styles from './Card.module.css';

const Card = ({ country }) => (
    <div data-testid='card'>
        <div className={styles.card}>
            <div data-testid='name'>Name: {country.name}</div>
            <div data-testid='confirmed'>Confirmed: {country.mostRecent.confirmed}</div>
            <div data-testid='deaths'>Deaths: {country.mostRecent.deaths}</div>
        </div>
    </div>
);

export default Card;