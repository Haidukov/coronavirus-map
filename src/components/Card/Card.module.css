.card {
    position: relative;
    left: -100px;
    background-color: white;
    font-size: 16px;
    min-width: 150px;
    margin: 12px;
    padding: 1rem;
    border-radius: 5px;
    color: #666;
    box-shadow: 0 2px 3px #d6d6d6;
    border: 1px solid #eee;
}