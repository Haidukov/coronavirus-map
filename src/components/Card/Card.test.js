import React from 'react';
import { render } from '@testing-library/react';
import Card from './Card';

describe('<Card/>', () => {
    it('should render data', () => {
        const countryMock = {
            name: 'Ukraine',
            mostRecent: {
                confirmed: 100,
                deaths: 3
            }
        };
        const { getByTestId, container } = render((
            <Card
                country={countryMock}
            />
        ));

        const name = getByTestId('name');
        const confirmed = getByTestId('confirmed');
        const deaths = getByTestId('deaths');
        
        expect(name).toHaveTextContent('Name: Ukraine');
        expect(confirmed).toHaveTextContent('Confirmed: 100');
        expect(deaths).toHaveTextContent('Deaths: 3');
        expect(container).toMatchSnapshot();
    });
});