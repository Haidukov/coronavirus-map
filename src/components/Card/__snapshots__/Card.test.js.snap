// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> should render data 1`] = `
<div>
  <div
    data-testid="card"
  >
    <div
      class="card"
    >
      <div
        data-testid="name"
      >
        Name: 
        Ukraine
      </div>
      <div
        data-testid="confirmed"
      >
        Confirmed: 
        100
      </div>
      <div
        data-testid="deaths"
      >
        Deaths: 
        3
      </div>
    </div>
  </div>
</div>
`;
